parsing/parser.mly

```diff
@@ -1816,6 +1810,8 @@ simple_pattern_not_ident:
       { $1 }
   | mod_longident DOT simple_delimited_pattern
       { mkpat @@ Ppat_open(mkrhs $1 1, $3) }
+  | mod_longident DOT LBRACKET RBRACKET
+    { mkpat @@ Ppat_construct ( mkrhs (Lident "[]") 1, None) }
```

Comment:
You are right `M.[]` should be the pattern open symmetric of ` M.[] ⇔ let open M in [] ` on the expression side. So, this is one difference between the two patch sets.